### DISCUSSION TITLE
Query about ViewModel and Factory Usage in ProyectTracker Project with Csla

### DISCUSSION THREAD

**Discussion by ecaldentey** [MarimerLLC/csla#4071] [Questions]
Hi Rocky,

In the ProyectTracker project, you use an object from the Csla.Blazor.ViewModel class to wrap the Csla BusinessClass objects, and through the ViewModel objects, you interact with the UI components. To establish the connection, you use the following syntax:

```c#
await vm.RefreshAsync(() => projectEditPortal.CreateAsync());
```

Since I have all my classes written using Criteria objects and go through my own Factory class of the BusinessClass, I use the following syntax to interact with my BusinessClass:

```c#
var dataportal = new CustomerEditList.Factory(_customerListFactory);

CustomerDto customerDto = new CustomerDto
{
    CustomerId = 2
};

_customerList = await dataportal.GetAsync(customerDto, CustomerSelectDbTypes.SelectAll);
```

Therefore, is it possible to load the ViewModel object directly with a BusinessClass (Root, EditList, etc.) and not use a factory?

In this same case, if the Razor page is created on the server-side, does the technique of detecting the rendering state that you use on the EditProject page make sense?

Any help will be appreciated, thank you very much.

Best regards

**Comment by rockfordlhotka** [comment]
The `RefreshAsync` method in the `ViewModel` class expects that you'll provide a method that returns a `Task<T>` where `T` is the business object type.

https://github.com/MarimerLLC/csla/blob/56fb6832547539b671a5ec98428625c68d73f8d2/Source/Csla.Blazor/ViewModel.cs#L194

If your factory method of your factory class returns a `Task<T>` you can have `RefreshAsync` method call your factory.
